### PR TITLE
Add OpenMP dataiterator test to unit tests

### PR DESCRIPTION
**Pull request by dschwoerer**
Is this correct?
Should I make sure the test is run in parallel with an acceptable number of processors?

### CODE REVIEW

File: tests/unit/field/test_field3d.cxx [hunk 1]
```diff
@@ -1239,3 +1239,36 @@ TEST_F(Field3DTest, DC) {
...
+    d3[i]=1;
+  }
+
+  BoutReal exp=1;
+
+  BOUT_OMP(parallel for)
+  for (int j=0;j<fields;++j){
+    BOUT_OMP(parallel)
+    for (auto i: d3[j]){
+      EXPECT_TRUE(d3[j][i] == exp);
```
Comment (ZedThree) [inline tests/unit/field/test_field3d.cxx]: You should be able to use `EXPECT_DOUBLE_EQ(d3[j][i], exp)` here, which compares floats with some tolerance

File: tests/unit/field/test_field3d.cxx [hunk 2]
```diff
@@ -1239,3 +1239,36 @@ TEST_F(Field3DTest, DC) {
...
+
+#ifdef _OPENMP
+TEST_F(Field3DTest, OpenMPIterator) {
+  const int fields=10;
+  Field3D * d3= new Field3D[fields];
+  for (int i=0;i<fields;++i){
+    d3[i]=1;
+  }
+
+  BoutReal exp=1;
```
Comment (ZedThree) [inline tests/unit/field/test_field3d.cxx]: `exp` -> `expected`

Non-abbreviated variable names are good :)

**Comment by ZedThree** [comment]
It'll get run with a couple of threads in the OpenMP build, that's probably sufficient for now. When we've got more OpenMP-specific tests in there, we should run the unit tests at least with a different numbers of threads.

Not required, but if you've got clang-format installed, could you run it over this test?

**Comment by dschwoerer** [comment]
I should have addressed all 3 changes.
Thanks for the feedback :+1:

**Pull request merged**